File: skinet.client/src/app/api/model/shopCart.ts

import { CartItem } from './cartItem';
import { nanoid } from "nanoid"

export class ShopCart implements CartType {
  id = nanoid();
  items: CartItem[] = [];
  deliverymethodId?: number;
  PaymentId?: string;
  clientSecret?: string;
}

export interface CartType {
  id: string;
  items: CartItem[]
  deliverymethodId?: number;
  PaymentId?: string;
  clientSecret?: string;
}

File: SKINET.Server/Entities/ShopCart.cs
namespace SKINET.Server.Entities
{
    public class ShopCart
    {
        public required string Id { get; set; }  
        public List<CartItem> Items { get; set; } = [];
        public int? DeliverymethodId { get; set; }
        public string? ClientSecret { get; set; }
        public string? PaymentmethodId { get; set; }

    }
}


